### ISSUE TITLE
Documentation improvements

### ISSUE THREAD

**Issue by ryanchristo** [regen-network/regen-ledger#889]
## Summary

Currently the module documentation is using the same structure and format as [specified in the Cosmos SDK](https://github.com/cosmos/cosmos-sdk/blob/master/docs/spec/SPEC-SPEC.md) but also lacking on both "robustness" and "readability/accessibility". We need to make improvements that provide a better user experience for developers and technically-minded users.

Aside from updating the module documentation to include functionality that has already been implemented but is not fully documented, what improvements can we make to our module documentation? This issue can be used as a starting point for further discussion and any specific issues should be broken out from here.

## Problem Definition

We have received feedback that our documentation lacks "robustness" and "readability/accessibility".

## Proposal

[work in progress]

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by ryanchristo** [comment]
Notes from regen-ledger product call:

- The landing page "overview" should cater to a non-technical audience.
- We should avoid duplicating anything that is already generated in the buf docs.
- We need to include generated docs for "features" that can be auto-generated.
- CLI commands and documentation should be auto-generated including help text.
- Make sure we are not duplicating work with https://library.regen.network/ and linking to these docs.
